### DISCUSSION TITLE
How to color categories

### DISCUSSION THREAD

**Discussion by pietru2004** [bitbrain/pandora#107] [Q&A] [answered]
I want to ask how to color those categories like in image from readme
![obraz](https://github.com/bitbrain/pandora/assets/51274570/b1f3a966-db09-4916-a14d-a418bc534231)

Also is there any demo for categories? (Well icons from readme img look nice)

Also this is what I got
![obraz](https://github.com/bitbrain/pandora/assets/51274570/a00f0a64-4799-4c96-aadf-ba30bb6f2058)

**Comment by bitbrain** [comment]
Hey @pietru2004 - these are icons downloaded from [game-icons.net](https://game-icons.net) and then manually coloured via [Inkscape](https://inkscape.org/release/inkscape-1.3/). That being said, this can be very complicated and colouring icons automatically would be quite useful. Also maybe having a set of "icons" to choose from (with color picker) might be a cool feature for this addon!

**Reply by pietru2004** [reply]
so you could just do this using self modulate on icon right ?

**Reply by pietru2004** [reply]
also I heard of game icons... I didn't know icons were from it :)

**Reply by bitbrain** [reply, marked as answer]
> so you could just do this using self modulate on icon right?

Oh wow, good shout! I discovered there is https://docs.godotengine.org/en/stable/classes/class_treeitem.html#class-treeitem-method-set-icon-modulate

This can be implemented very easily:

1. add new `color` attribute and make it configurable to the entity configuration (point 5 on https://bitbra.in/pandora/#/concepts/editor)
2. make `color` inherit from parent by default unless overridden
3. change logic in entity_tree.gd to set the modulate item color based on the color of the entity https://github.com/bitbrain/pandora/blob/godot-4.x/addons/pandora/ui/components/entity_tree/entity_tree.gd#L130
4. setup a signal to when the color attribute changes to auto-update the icons in the tree

**Comment by bitbrain** [comment]
@pietru2004 created https://github.com/bitbrain/pandora/issues/110
